packages/components/src/components/navigation/actions.ts

```diff
@@ -1,18 +1,21 @@
-import { Ajax, Utils } from '@labkey/api';
+import { Ajax, Utils, ActionURL } from '@labkey/api';
 import { buildURL } from "../../url/ActionURL";
 
-export function signOut() {
+export function signOut(navigateUrl?: string) {
     const startUrl = buildURL('project', 'start', undefined, {returnURL: false});
 
+    // for the redirectUrl to work in the case of CAS logout provider redirect, this URL needs to include the host (Issue 39803)
+    const returnUrl = ActionURL.getBaseURL(true) + navigateUrl || startUrl;
+
     Ajax.request({
-        url: buildURL('login', 'logoutAPI.api'),
+        url: buildURL('login', 'logoutAPI.api', undefined, {returnURL: returnUrl}),
         method: 'POST',
         success: Utils.getCallbackWrapper((response) => {
-            window.location.href = startUrl;
+            window.location.href = response.redirectUrl ? response.redirectUrl : returnUrl;
```

Comment:
```suggestion
            window.location.href = response.redirectUrl || returnUrl;
```

Reply:
Yup, good suggestion. Will make that change.